zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -35,7 +35,10 @@ pub enum InputSelectorError<DbErrT, SelectorErrT> {
     Selection(SelectorErrT),
     /// Insufficient funds were available to satisfy the payment request that inputs were being
     /// selected to attempt to satisfy.
-    InsufficientFunds { available: Amount, required: Amount },
+    InsufficientFunds {
+        available: NonNegativeAmount,
+        required: NonNegativeAmount,
+    },
```

Comment:
This type seems to be essentially a subset of `data_api::error::Error`. Can the duplication be reduced somehow?